### PR TITLE
Add `rel="noopener noreferrer"` to `target="_blank"` links

### PR DESCRIPTION
**Pull request by ClementChaumel**
## Done

Links opening in a new tab (with attribute `target="_blank") are a potential vulnerability. 
More details and examples here: https://mathiasbynens.github.io/rel-noopener/
Adding `rel="noopener noreferrer"` fix that

## QA

- Open the demo site on a page which have a target="_blank" link such as [here](https://ubuntu-com-9146.demos.haus/wsl#:~:text=Discover%20how%20to%20easily%20access%20“Kubernetes%20on%20Windows%20with%20WSL2”%20in%20our%20webinar) 
- Verify that it still works as expected

## Issue / Card

Fixes #

## Screenshots

[If relevant, please include a screenshot.]

### CODE REVIEW

**Review by tbille — CHANGES_REQUESTED** [review]
`noreferrer` is useful for user generated content, which is not the case here. I don't think we should use it in this scenario because this will have an SEO impact.

`noreferrer` is a way to signal that maybe this site is linked on our site but we don't vouch for it.

It feels in this case `noopener` should be enough. What do you think?

File: templates/legal/terms-and-policies/contact-us.html [hunk 1]
```diff
@@ -661,7 +661,7 @@ <h1>Trademark enquiries</h1>
                     <div class="oneField field-container-D  labelsAbove  " id="tfa_Pleaseaddanyothe-D">
                       <label id="tfa_Pleaseaddanyothe-L" class="label preField reqMark" for="tfa_Pleaseaddanyothe">Please describe your request in more detail</label><br><div class="inputWrapper"><textarea cols="90" rows="6" aria-required="true" id="tfa_Pleaseaddanyothe" name="tfa_Pleaseaddanyothe" title="Please describe your request in more detail" class="required"></textarea></div>
                     </div>
-                    <div class="htmlSection" id="tfa_9631608629066"><div class="htmlContent" id="tfa_9631608629066-HTML"><font color="#a4a4a4">Data collected from this survey will be handled in accordance with the Canonical <a target="_blank" href="http://www.ubuntu.com/legal">privacy policy</a></font></div></div>
+                    <div class="htmlSection" id="tfa_9631608629066"><div class="htmlContent" id="tfa_9631608629066-HTML"><font color="#a4a4a4">Data collected from this survey will be handled in accordance with the Canonical <a target="_blank" rel="noopener noreferrer" href="http://www.ubuntu.com/legal">privacy policy</a></font></div></div>
```
Comment (anthonydillon) [inline templates/legal/terms-and-policies/contact-us.html:664]: This one should be relative and remove the blank.

File: templates/legal/contributors/agreement.html [hunk 2]
```diff
@@ -332,7 +332,7 @@ <h2 class="western">6. Miscellaneous</h2>
...
                       <p>6.5 If any provision of this Agreement is found void and unenforceable, such provision will be replaced to the extent possible with a provision that comes closest to the meaning of the original provision and which is enforceable. The terms and conditions set forth in this Agreement shall apply notwithstanding any failure of essential purpose of this Agreement or any limited remedy to the maximum extent possible under law.</p>
-                      <p>To <a target="_blank" href="http://www.canonical.com/sites/default/files/active/images/Canonical-HA-CLA-ANY-E.pdf">download as a PDF</a>
+                      <p>To <a target="_blank" rel="noopener noreferrer" href="http://www.canonical.com/sites/default/files/active/images/Canonical-HA-CLA-ANY-E.pdf">download as a PDF</a>
```
Comment (anthonydillon) [inline templates/legal/contributors/agreement.html:335]: Does this link work? If so, can you update the link and remove the blank please.

File: templates/legal/contributors/agreement.html [hunk 3]
```diff
@@ -277,7 +277,7 @@ <h2 class="western">6. Miscellaneous</h2>
...
                         <p>6.5 If any provision of this Agreement is found void and unenforceable, such provision will be replaced to the extent possible with a provision that comes closest to the meaning of the original provision and which is enforceable. The terms and conditions set forth in this Agreement shall apply notwithstanding any failure of essential purpose of this Agreement or any limited remedy to the maximum extent possible under law.</p>
-                        <p>To <a target="_blank" href="http://www.canonical.com/sites/default/files/active/images/Canonical-HA-CLA-ANY-I.pdf">download as a PDF</a>
+                        <p>To <a target="_blank" rel="noopener noreferrer" href="http://www.canonical.com/sites/default/files/active/images/Canonical-HA-CLA-ANY-I.pdf">download as a PDF</a>
```
Comment (anthonydillon) [inline templates/legal/contributors/agreement.html:280]: Update the link and remove blank

File: templates/legal/confidentiality-agreement.html [hunk 4]
```diff
@@ -498,7 +498,7 @@ <h3>Your details</h3>
                     </li>
                     <li class="p-list__item">
                       <div class="htmlSection" id="tfa_5">
-                        <p>By clicking "I agree" below, I confirm that I have read and understood the terms of the confidentiality agreement, <a href="/legal/data-privacy">privacy policy</a> and <a href="/legal/data-privacy/unilateral-nda">privacy notice</a> and warrant that I have authority to enter into the <a href="https://assets.ubuntu.com/v1/05d55fbc-Canonical+NDA_Online+Form_202008.pdf" target="_blank">confidentiality agreement</a>
+                        <p>By clicking "I agree" below, I confirm that I have read and understood the terms of the confidentiality agreement, <a href="/legal/data-privacy">privacy policy</a> and <a href="/legal/data-privacy/unilateral-nda">privacy notice</a> and warrant that I have authority to enter into the <a href="https://assets.ubuntu.com/v1/05d55fbc-Canonical+NDA_Online+Form_202008.pdf" target="_blank" rel="noopener noreferrer">confidentiality agreement</a>
```
Comment (anthonydillon) [inline templates/legal/confidentiality-agreement.html:501]: I don't think this needs blank as the user have the back button if they need. They are not missing any state.

File: templates/legal/confidentiality-agreement.html [hunk 5]
```diff
@@ -115,7 +115,7 @@
   <div class="row">
     <div class="col-8">
       <h1>Confidentiality agreement</h1>
-      <p>You can enter into a <a href="https://assets.ubuntu.com/v1/05d55fbc-Canonical+NDA_Online+Form_202008.pdf" target="_blank">confidentiality agreement</a> with Canonical here, you must complete all the fields in this form.</p>
+      <p>You can enter into a <a href="https://assets.ubuntu.com/v1/05d55fbc-Canonical+NDA_Online+Form_202008.pdf" target="_blank" rel="noopener noreferrer">confidentiality agreement</a> with Canonical here, you must complete all the fields in this form.</p>
```
Comment (anthonydillon) [inline templates/legal/confidentiality-agreement.html:118]: Revive blank please

File: templates/security/cve/_cve-table.html [hunk 6]
```diff
@@ -91,10 +91,10 @@
                 {% if statuses[release.codename].pocket != "esm-infra" and statuses[release.codename].pocket == "esm-apps" %}
                   <small class="cve-table-cell esm-info">
                     {% if statuses[release.codename].pocket == "esm-infra" %}
-                      <a href="/advantage" target="_blank">UA Infra/Apps</a>
+                      <a href="/advantage" target="_blank" rel="noopener noreferrer">UA Infra/Apps</a>
```
Comment (anthonydillon) [inline templates/security/cve/_cve-table.html:94]: I don't know why this was blank, can you remove it please.

File: templates/security/cve/_cve-table.html [hunk 7]
```diff
@@ -91,10 +91,10 @@
...
                     {% endif %}
                     {% if statuses[release.codename].pocket == "esm-apps" %}
-                      <a href="/advantage" target="_blank">UA Apps</a>
+                      <a href="/advantage" target="_blank" rel="noopener noreferrer">UA Apps</a>
```
Comment (anthonydillon) [inline templates/security/cve/_cve-table.html:97]: Same here

File: templates/security/cve/cve.html [hunk 8]
```diff
@@ -137,11 +137,11 @@ <h2>Status</h2>
                   {% if release.codename in statuses %}
                     {% if loop.index == 1 %}
                       <td rowspan="{{ releases | length }}">
-                        <a href="/security/cve?q=&package={{ package_name }}" target="_blank">{{ package_name }}</a><br>
+                        <a href="/security/cve?q=&package={{ package_name }}" target="_blank" rel="noopener noreferrer">{{ package_name }}</a><br>
```
Comment (anthonydillon) [inline templates/security/cve/cve.html:140]: Again this is simply a link from the listing page to a details.

**Comment by ClementChaumel** [comment]
> `noreferrer` is useful for user generated content, which is not the case here. I don't think we should use it in this scenario because this will have an SEO impact.
> 
> `noreferrer` is a way to signal that maybe this site is linked on our site but we don't vouch for it.
> 
> It feels in this case `noopener` should be enough. What do you think?

It shouldn't really be an issue anymore since most browsers support it now but my concern was that if someone is using an outdated browser `noreferrer` would act as a fallback if `noopener` was not supported.

And I'm no SEO wizard but from what I could find, neither of them seem to have an impact on SEO : https://www.searchcandy.uk/seo/does-adding-relnoopener-noreferrer-to-a-link-affect-seo/#Does-adding-relnoopener-noreferrer-to-links-have-any-impact-on-SEO

File: templates/security/cve/cve.html [hunk 9]
```diff
@@ -137,11 +137,11 @@ <h2>Status</h2>
...
                         <small>
-                        <a href="https://launchpad.net/distros/ubuntu/+source/{{ package_name }}" target="_blank">Launchpad</a>,
-                        <a href="https://packages.ubuntu.com/search?suite=all&section=all&arch=any&searchon=sourcenames&keywords={{ package_name }}" target="_blank">Ubuntu</a>,
-                        <a href="https://tracker.debian.org/{{ package_name }}" target="_blank">Debian</a>
+                        <a href="https://launchpad.net/distros/ubuntu/+source/{{ package_name }}" target="_blank" rel="noopener noreferrer">Launchpad</a>,
+                        <a href="https://packages.ubuntu.com/search?suite=all&section=all&arch=any&searchon=sourcenames&keywords={{ package_name }}" target="_blank" rel="noopener noreferrer">Ubuntu</a>,
+                        <a href="https://tracker.debian.org/{{ package_name }}" target="_blank" rel="noopener noreferrer">Debian</a>
```
Comment (anthonydillon) [inline templates/security/cve/cve.html:144]: Why would any of these be blank. Is there state the user would lose by navigating away at this point?

File: templates/security/cve/cve.html [hunk 10]
```diff
@@ -209,13 +209,13 @@ <h2>Status</h2>
                           {{ patch.content }}
                         {% elif patch.type == "break-fix" %}
                           Introduced by
-                          <a href="https://git.kernel.org/linus/{{ patch.content.introduced }}" target="_blank">{{ patch.content.introduced }}</a><br>
+                          <a href="https://git.kernel.org/linus/{{ patch.content.introduced }}" target="_blank" rel="noopener noreferrer">{{ patch.content.introduced }}</a><br>
```
Comment (anthonydillon) [inline templates/security/cve/cve.html:212]: I hope this is to try and retain users when they have indicated they want to follow a link.

File: templates/security/cve/cve.html [hunk 11]
```diff
@@ -209,13 +209,13 @@ <h2>Status</h2>
...
                           Fixed by
-                          <a href="https://git.kernel.org/linus/{{ patch.content.fixed }}" target="_blank">{{ patch.content.fixed }}</a>
+                          <a href="https://git.kernel.org/linus/{{ patch.content.fixed }}" target="_blank" rel="noopener noreferrer">{{ patch.content.fixed }}</a>
                         {% elif patch.type == "link" %}
                           {{ patch.content.prefix }}:
                           {% if cve._clean_url(patch.content.suffix) %}
-                            <a href="{{ cve._clean_url(patch.content.suffix) }}" target="_blank">{{ patch.content.suffix }}</a>
+                            <a href="{{ cve._clean_url(patch.content.suffix) }}" target="_blank" rel="noopener noreferrer">{{ patch.content.suffix }}</a>
```
Comment (anthonydillon) [inline templates/security/cve/cve.html:218]: Same here. I believe this can be a normal link

**Comment by tbille** [comment]
@ClementChaumel  you are right for a reason I confused `rel=ugc` and `no=referrer`. My mistake I used my memory, I shouldn't :laughing:

File: templates/shared/contextual_footers/_devices_newsletter_signup.html [hunk 12]
```diff
@@ -10,7 +10,7 @@
         @media only screen and (min-width: 768px) { #mce-success-response { height: 90%; }}
         #mc_embed_signup input.mce_inline_error { border-color: red; }
       </style>
-      <form action="https://canonical.us3.list-manage.com/subscribe/post?u=56dac47c206ba0f58ec25f314&amp;id=97a67dea26" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" class="mc-embedded-subscribe-form validate" target="_blank" novalidate>
+      <form action="https://canonical.us3.list-manage.com/subscribe/post?u=56dac47c206ba0f58ec25f314&amp;id=97a67dea26" method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" class="mc-embedded-subscribe-form validate" target="_blank" rel="noopener noreferrer" novalidate>
```
Comment (anthonydillon) [inline templates/shared/contextual_footers/_devices_newsletter_signup.html:13]: This seems validate but very awkward. This should be replaced with a returning post or an background submit.

**Review by tbille — APPROVED** [review]

File: templates/whitepapers/beta/shared/form.html [hunk 13]
```diff
@@ -44,7 +44,7 @@
       </p>
     </li>
     <li  class="mktField p-list__item">
-      <p>In submitting this form, I confirm that I have read and agree to <a href="https://www.ubuntu.com/legal/dataprivacy" target="_blank" class="mchNoDecorate" id="">Canonical's Privacy Policy</a></p>
+      <p>In submitting this form, I confirm that I have read and agree to <a href="https://www.ubuntu.com/legal/dataprivacy" target="_blank" rel="noopener noreferrer" class="mchNoDecorate" id="">Canonical's Privacy Policy</a></p>
```
Comment (anthonydillon) [inline templates/whitepapers/beta/shared/form.html:47]: Relative and remove blank please

File: templates/whitepapers/shared/form.html [hunk 14]
```diff
@@ -44,7 +44,7 @@
       </p>
     </li>
     <li  class="mktField p-list__item">
-      <p>In submitting this form, I confirm that I have read and agree to <a href="https://www.ubuntu.com/legal/dataprivacy" target="_blank" class="mchNoDecorate" id="">Canonical's Privacy Policy</a></p>
+      <p>In submitting this form, I confirm that I have read and agree to <a href="https://www.ubuntu.com/legal/dataprivacy" target="_blank" rel="noopener noreferrer" class="mchNoDecorate" id="">Canonical's Privacy Policy</a></p>
```
Comment (anthonydillon) [inline templates/whitepapers/shared/form.html:47]: Relative and remove blank please

File: templates/wsl/index.html [hunk 15]
```diff
@@ -60,12 +60,12 @@ <h3 class="p-heading--four">Access the Linux terminal on Windows</h3>
     <div class="col-4 p-divider__block">
       <h3 class="p-heading--four">Develop cross-platform</h3>
       <p>Build and debug Linux applications  with Windows tools like Visual Studio Code, Visual Studio, and JetBrains IDEs before deploying to the cloud.</p>
-      <p><a class="p-link--external" href="https://youtu.be/DmfuJzX6vJQ" target="_blank">Discover how to easily access "Kubernetes on Windows with WSL2" in our webinar</a></p>
+      <p><a class="p-link--external" href="https://youtu.be/DmfuJzX6vJQ" target="_blank" rel="noopener noreferrer">Discover how to easily access "Kubernetes on Windows with WSL2" in our webinar</a></p>
```
Comment (anthonydillon) [inline templates/wsl/index.html:63]: This is simply a link to a video from a static page. If you want to retain people on the page embed the video

File: templates/wsl/index.html [hunk 16]
```diff
@@ -60,12 +60,12 @@ <h3 class="p-heading--four">Access the Linux terminal on Windows</h3>
...
     </div>
     <div class="col-4 p-divider__block">
       <h3 class="p-heading--four">Manage IT infrastructure</h3>
       <p>From the same workstation, manage mixed Linux and Windows infrastructure both on-prem and across public clouds.</p>
-      <p><a class="p-link--external" href="https://youtu.be/50bokFFOvtA" target="_blank">Learn from our community members and experts at the WSLConf</a></p>
+      <p><a class="p-link--external" href="https://youtu.be/50bokFFOvtA" target="_blank" rel="noopener noreferrer">Learn from our community members and experts at the WSLConf</a></p>
```
Comment (anthonydillon) [inline templates/wsl/index.html:68]: Same here

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Generally the use of `_blank` on links is an anti-pattern. Which it looks like has been used here in the past. If a user is not losing state or position in a SPA then the use to blank should be discouraged. User expect links to change the current tab which does not break the back button.

File: templates/legal/contributors/agreement.html (hunk 2 above)
Comment (ClementChaumel) [reply templates/legal/contributors/agreement.html]: Both those links are broken. Do you know who I should ask to get proper ones?

**Review by anthonydillon — APPROVED** [review]
LGTM :+1: thanks

**Pull request merged**